tiledb/sm/compressors/dd_compressor.cc

```diff
@@ -111,6 +111,10 @@ Status DoubleDelta::compress(
       return LOG_STATUS(Status_CompressionError(
           "Cannot compress tile with DoubleDelta; Float "
           "datatypes are not supported"));
+    case Datatype::BOOL:
```

Comment:
I don't think this restriction is necessary given we can use doubledelta on uint8?